proof/refine/AARCH64/Retype_R.thy

```diff
@@ -1329,8 +1327,10 @@ lemma retype_state_relation:
     have [simp]: "cns_of_heap ?ps = cns_of_heap (kheap s)"
       by - (rule ext, induct (?al),
             simp_all add: cns_of_heap_def default_object_def TCBObject)
+    from pn2
     have [simp]: "pt_types_of_heap ?ps = pt_types_of_heap (kheap s)"
-      sorry (* FIXME AARCH64 *)
+      by - (rule ext, induct (?al),
+            simp_all add: pt_types_of_heap_def default_object_def TCBObject opt_map_def)
```

Comment:
Super rarely used. Usually if you give a term instead of a variable to `induct`, it will generate equations of the form `x == t` where `x` is a fresh variable, and then do induction on `x`. If you really want to instantiate the induction rule with the term instead, then you can put the term in parentheses.

Usually the induction rules are not set up to work that way, so there must be some special case going on in this proof where `?al` works out to be something useful for the induction rule.